docs/jwst/refpix/arguments.rst

```diff
@@ -52,3 +52,30 @@ in IRS2 mode will be processed along with the normal pixels and preserved
 in the output.  This option is intended for calibration or diagnostic reductions
 only. For normal science operation, this argument should always be False,
 so that interleaved pixels are stripped before continuing processing.
+
+*  ``--use_conv_kernel``
+
+If the ``use_conv_kernel`` argument is set to False, all NIR full-frame data,
+will be processed using the running median and the side-pixel correction. The
+default value is set to True, which turns off the side-pixed correction and
+use an optimized convolution kernel instead of the running median.
+
+*  ``--sigreject``
+
+The ``sigreject`` argument is the number of sigmas to reject as outliers in the
+optimized convolution kernel algorithm. The value is expected to be a float.
+
+*  ``--gaussmooth``
+
+The ``gaussmooth`` argument is the width of Gaussian smoothing kernel to use as
+a low-pass filter. The numerical value is expected to be a float.
+
+*  ``--halfwidth``
+
+The ``halfwidth`` argument is the half-width of convolution kernel to build. The
+numerical value is expected to be an integer.
+
+*  ``--user_supplied_reffile``
+
+The ``user_supplied_reffile`` argument is the name of the ASDF user-supplied
+reference file for the optimized convolution kernel algorithm.
```

Comment:
This parameter is redundant with ``override_refpix``. Users who wish to supply a reference file should use the built-in method with the override parameter; the code will need some small updates to accommodate.

Reply:
True; that raises an interesting question though. NIRSpec IRS2 modes already use a 'refpix' fits-type reference file in CRDS.  The new reference file for non-IRS2 modes is type conv_kernel, so presumably the step will have to be able to handle both types of reference file inputs, and would need override_conv_kernel instead?

Reply:
My half-formed first thought was that this new file was going into the `refpix` type, but clearly that doesn't make sense... Yes, the step will need its list of reftypes expanded to include the new reftype. The override_{} parameters are generated from the list of step.reference_file_types, so an override parameter will be generated once e.g. `conv_kernel` is added to the list.